src/main/java/at/favre/lib/crypto/HkdfMacFactory.java

```diff
@@ -107,13 +130,25 @@ public Mac createInstance(byte[] key) {
                     hmacInstance = Mac.getInstance(macAlgorithmName, provider);
                 }
 
-                hmacInstance.init(secretKey);
                 return hmacInstance;
             } catch (NoSuchAlgorithmException e) {
                 throw new IllegalStateException("defined mac algorithm was not found", e);
             } catch (Exception e) {
-                throw new IllegalStateException("could not make hmac hasher in hkdf", e);
+                throw new IllegalStateException("could not create mac instance in hkdf", e);
+            }
+        }
+
+        @Override
+        public int getMacLengthBytes() {
+            return createMacInstance().getMacLength();
+        }
+
+        @Override
+        public SecretKey createSecretKey(byte[] rawKeyMaterial) {
+            if (rawKeyMaterial == null || rawKeyMaterial.length <= 0) {
```

Comment:
I think `rawKeyMaterial.length == 0` should be sufficient

Reply:
Yes. Altough I would also prefer a fail fast, null is an option here.